### PR TITLE
[DOCS] Reformat `stop` token filter

### LINKED ISSUE #16561: No documentation on what stopwords are in each set
For instance, there is no documentation of what words are in the `_english_` stop words set.

### PR DESCRIPTION
**Pull request by jrodewig**
Makes the following changes to the `stop` token filter docs:

* Updates description
* Adds a link to the related Lucene filter
* Adds detailed analyze snippet
* Updates custom analyzer and custom filter snippets
* Adds a list of predefined stop words by language

Closes #16561.

### CODE REVIEW

File: docs/reference/analysis/tokenfilters/stop-tokenfilter.asciidoc [hunk 1]
```diff
@@ -4,79 +4,374 @@
...
+
+[[analysis-stop-tokenfilter-configure-parms]]
+==== Configurable parameters
 
-[horizontal]
 `stopwords`::
++
+--
+(Optional, string or array of strings)
+Language value, such as `_arabic_` or `thai`. Defaults to
```
Comment (romseygeek) [inline docs/reference/analysis/tokenfilters/stop-tokenfilter.asciidoc]: s/`thai`/`_thai_`/

**Review by romseygeek — APPROVED** [review]
One nit, LGTM otherwise

File: docs/reference/analysis/tokenfilters/stop-tokenfilter.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/analysis/tokenfilters/stop-tokenfilter.asciidoc]: Good catch. Thanks!

**Comment by jrodewig** [comment]
### Backport commits
master e016864
7.x cf87724
7.6 c46c628

**Pull request merged**